File: src/typegraph/deno/src/runtimes/grpc.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { Func } from "../types.ts";
import { runtimes } from "../sdk.ts";
import { Runtime } from "./mod.ts";

export class GrpcRuntime extends Runtime {
  constructor(protoFile: string, endpoint: string) {
    const id = runtimes.registerGrpcRuntime({
      protoFile,
      endpoint,
    });
    super(id);
  }

  call(method: string) {
    const funcData = runtimes.callGrpcMethod(this._id, { method: method });
    return Func.fromTypeFunc(funcData);
  }
}

File: src/typegraph/schema/src/runtimes/grpc.rs
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct GrpcRuntimeData {
    pub proto_file_content: String,
    pub endpoint: String,
}
